CloudWatchWriter causes exceptions when running outside us-east-1

We use Cloudformation/Terraform to launch Zookeeper clusters. This is our repo: https://github.com/LoyaltyOne/bazooka.

We can use jmxtrans successfully when running the clusters in us-east-1. However, when running in ca-central-1 we started seeing the following exception:

```
java.lang.NullPointerException: Problems getting AWS metadata
```
I have tried version 265 and 267 with the same results.

This is the configuration file we use for our EC2 instances:
```json
{
  "servers": [
    {
      "alias": "ip-10-0-1-225.ca-central-1.compute.internal",
      "host": "localhost",
      "port": "8989",
      "numQueryThreads" : 2,
      "queries": [
        {
          "obj": "org.apache.ZooKeeperService:name0=ReplicatedServer_id*",
          "resultAlias": "Zookeeper_Server",
          "attr": ["QuorumSize"],
          "outputWriters": [
            {
              "@class": "com.googlecode.jmxtrans.model.output.CloudWatchWriter",
              "settings": {
                "namespace": "Zookeeper",
                "dimensions": [
                  {"name": "InstanceId", "value": "i-09e144cd7392ff51a"}
                ]
              }
            }
          ]
        },
        {
          "obj": "org.apache.ZooKeeperService:name0=ReplicatedServer_id*,name1=*,name2=*",
          "resultAlias": "Zookeeper_Clients",
          "attr": [
            "MaxClientCnxnsPerHost",
            "MaxRequestLatency",
            "AvgRequestLatency",
            "MinRequestLatency",
            "MaxSessionTimeout",
            "MinSessionTimeout",
            "OutstandingRequests",
            "PacketsReceived",
            "PacketsSent",
            "PendingRevalidationCount",
            "TickTime",
            "NumAliveConnections"
          ],
          "outputWriters": [
            {
              "@class": "com.googlecode.jmxtrans.model.output.CloudWatchWriter",
              "settings": {
                "namespace": "Zookeeper",
                "dimensions": [
                  {"name": "InstanceId", "value": "i-09e144cd7392ff51a"}
                ]
              }
            }
          ]
        },
        {
          "obj": "org.apache.ZooKeeperService:name0=ReplicatedServer_id*,name1=*,name2=*,name3=InMemoryDataTree",
          "resultAlias": "Zookeeper_InMemoryDataTree",
          "attr": ["NodeCount", "WatchCount"],
          "outputWriters": [
            {
              "@class": "com.googlecode.jmxtrans.model.output.CloudWatchWriter",
              "settings": {
                "namespace": "Zookeeper",
                "dimensions": [
                  {"name": "InstanceId", "value": "i-09e144cd7392ff51a"}
                ]
              }
            }
          ]
        },
        {
          "obj": "java.lang:type=Memory",
          "resultAlias": "JVM_Memory",
          "attr": ["HeapMemoryUsage", "NonHeapMemoryUsage"],
          "outputWriters": [
            {
              "@class": "com.googlecode.jmxtrans.model.output.CloudWatchWriter",
              "settings": {
                "namespace": "Zookeeper",
                "dimensions": [
                  {"name": "InstanceId", "value": "i-09e144cd7392ff51a"}
                ]
              }
            }
          ]
        },
        {
          "obj": "java.lang:type=Threading",
          "resultAlias": "JVM_Threading",
          "attr": ["ThreadCount"],
          "outputWriters": [
            {
              "@class": "com.googlecode.jmxtrans.model.output.CloudWatchWriter",
              "settings": {
                "namespace": "Zookeeper",
                "dimensions": [
                  {"name": "InstanceId", "value": "i-09e144cd7392ff51a"}
                ]
              }
            }
          ]
        }
      ]
    }
  ]
}
```
This is the entire log file:
```
STATUS | wrapper  | 2017/11/12 04:43:57 | --> Wrapper Started as Daemon
STATUS | wrapper  | 2017/11/12 04:43:57 | Launching a JVM...
INFO   | jvm 1    | 2017/11/12 04:43:58 | Wrapper (Version 3.2.3) http://wrapper.tanukisoftware.org
INFO   | jvm 1    | 2017/11/12 04:43:58 |   Copyright 1999-2006 Tanuki Software, Inc.  All Rights Reserved.
INFO   | jvm 1    | 2017/11/12 04:43:58 | 
[12 Nov 2017 04:43:58] [WrapperSimpleAppMain] 0      INFO  (com.googlecode.jmxtrans.JmxTransformer:181) - Starting Jmxtrans on : /var/lib/jmxtrans
[12 Nov 2017 04:43:58] [WrapperSimpleAppMain] 0      INFO  (com.googlecode.jmxtrans.JmxTransformer:181) - Starting Jmxtrans on : /var/lib/jmxtrans
[12 Nov 2017 04:43:59] [WrapperSimpleAppMain] 703    ERROR (com.googlecode.jmxtrans.JmxTransformer:190) - Error parsing json: /var/lib/jmxtrans/jmxtrans_zookeeper.json
com.googlecode.jmxtrans.exceptions.LifecycleException: Error parsing json: /var/lib/jmxtrans/jmxtrans_zookeeper.json
	at com.googlecode.jmxtrans.ConfigurationParser.parseServers(ConfigurationParser.java:62)
	at com.googlecode.jmxtrans.JmxTransformer.processFilesIntoServers(JmxTransformer.java:354)
	at com.googlecode.jmxtrans.JmxTransformer.startupSystem(JmxTransformer.java:322)
	at com.googlecode.jmxtrans.JmxTransformer.start(JmxTransformer.java:187)
	at com.googlecode.jmxtrans.JmxTransformer.doMain(JmxTransformer.java:157)
	at com.googlecode.jmxtrans.JmxTransformer.main(JmxTransformer.java:138)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.tanukisoftware.wrapper.WrapperSimpleApp.run(WrapperSimpleApp.java:240)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException: Problems getting AWS metadata
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:228)
	at com.googlecode.jmxtrans.model.output.CloudWatchWriter.createCloudWatchClient(CloudWatchWriter.java:97)
	at com.googlecode.jmxtrans.model.output.CloudWatchWriter.create(CloudWatchWriter.java:111)
	at com.googlecode.jmxtrans.model.SingletonOutputWriterFactory.<init>(SingletonOutputWriterFactory.java:40)
	at com.googlecode.jmxtrans.ServerListBuilder.singleton(ServerListBuilder.java:70)
	at com.googlecode.jmxtrans.ServerListBuilder.access$000(ServerListBuilder.java:46)
	at com.googlecode.jmxtrans.ServerListBuilder$TemporaryServer.addQuery(ServerListBuilder.java:103)
	at com.googlecode.jmxtrans.ServerListBuilder$TemporaryServer.addQueries(ServerListBuilder.java:94)
	at com.googlecode.jmxtrans.ServerListBuilder.add(ServerListBuilder.java:60)
	at com.googlecode.jmxtrans.ServerListBuilder.add(ServerListBuilder.java:53)
	at com.googlecode.jmxtrans.ConfigurationParser.parseServers(ConfigurationParser.java:57)
	... 11 more
[12 Nov 2017 04:43:59] [WrapperSimpleAppMain] 703    ERROR (com.googlecode.jmxtrans.JmxTransformer:190) - Error parsing json: /var/lib/jmxtrans/jmxtrans_zookeeper.json
com.googlecode.jmxtrans.exceptions.LifecycleException: Error parsing json: /var/lib/jmxtrans/jmxtrans_zookeeper.json
	at com.googlecode.jmxtrans.ConfigurationParser.parseServers(ConfigurationParser.java:62)
	at com.googlecode.jmxtrans.JmxTransformer.processFilesIntoServers(JmxTransformer.java:354)
	at com.googlecode.jmxtrans.JmxTransformer.startupSystem(JmxTransformer.java:322)
	at com.googlecode.jmxtrans.JmxTransformer.start(JmxTransformer.java:187)
	at com.googlecode.jmxtrans.JmxTransformer.doMain(JmxTransformer.java:157)
	at com.googlecode.jmxtrans.JmxTransformer.main(JmxTransformer.java:138)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.tanukisoftware.wrapper.WrapperSimpleApp.run(WrapperSimpleApp.java:240)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException: Problems getting AWS metadata
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:228)
	at com.googlecode.jmxtrans.model.output.CloudWatchWriter.createCloudWatchClient(CloudWatchWriter.java:97)
	at com.googlecode.jmxtrans.model.output.CloudWatchWriter.create(CloudWatchWriter.java:111)
	at com.googlecode.jmxtrans.model.SingletonOutputWriterFactory.<init>(SingletonOutputWriterFactory.java:40)
	at com.googlecode.jmxtrans.ServerListBuilder.singleton(ServerListBuilder.java:70)
	at com.googlecode.jmxtrans.ServerListBuilder.access$000(ServerListBuilder.java:46)
	at com.googlecode.jmxtrans.ServerListBuilder$TemporaryServer.addQuery(ServerListBuilder.java:103)
	at com.googlecode.jmxtrans.ServerListBuilder$TemporaryServer.addQueries(ServerListBuilder.java:94)
	at com.googlecode.jmxtrans.ServerListBuilder.add(ServerListBuilder.java:60)
	at com.googlecode.jmxtrans.ServerListBuilder.add(ServerListBuilder.java:53)
	at com.googlecode.jmxtrans.ConfigurationParser.parseServers(ConfigurationParser.java:57)
	... 11 more
INFO   | jvm 1    | 2017/11/12 04:43:59 | 
INFO   | jvm 1    | 2017/11/12 04:43:59 | WrapperSimpleApp: Encountered an error running main: com.googlecode.jmxtrans.exceptions.LifecycleException: com.googlecode.jmxtrans.exceptions.LifecycleException: Error parsing json: /var/lib/jmxtrans/jmxtrans_zookeeper.json
INFO   | jvm 1    | 2017/11/12 04:43:59 | com.googlecode.jmxtrans.exceptions.LifecycleException: com.googlecode.jmxtrans.exceptions.LifecycleException: Error parsing json: /var/lib/jmxtrans/jmxtrans_zookeeper.json
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.JmxTransformer.start(JmxTransformer.java:191)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.JmxTransformer.doMain(JmxTransformer.java:157)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.JmxTransformer.main(JmxTransformer.java:138)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at java.lang.reflect.Method.invoke(Method.java:498)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at org.tanukisoftware.wrapper.WrapperSimpleApp.run(WrapperSimpleApp.java:240)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at java.lang.Thread.run(Thread.java:748)
INFO   | jvm 1    | 2017/11/12 04:43:59 | Caused by: com.googlecode.jmxtrans.exceptions.LifecycleException: Error parsing json: /var/lib/jmxtrans/jmxtrans_zookeeper.json
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.ConfigurationParser.parseServers(ConfigurationParser.java:62)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.JmxTransformer.processFilesIntoServers(JmxTransformer.java:354)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.JmxTransformer.startupSystem(JmxTransformer.java:322)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.JmxTransformer.start(JmxTransformer.java:187)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	... 8 more
INFO   | jvm 1    | 2017/11/12 04:43:59 | Caused by: java.lang.NullPointerException: Problems getting AWS metadata
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:228)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.model.output.CloudWatchWriter.createCloudWatchClient(CloudWatchWriter.java:97)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.model.output.CloudWatchWriter.create(CloudWatchWriter.java:111)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.model.SingletonOutputWriterFactory.<init>(SingletonOutputWriterFactory.java:40)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.ServerListBuilder.singleton(ServerListBuilder.java:70)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.ServerListBuilder.access$000(ServerListBuilder.java:46)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.ServerListBuilder$TemporaryServer.addQuery(ServerListBuilder.java:103)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.ServerListBuilder$TemporaryServer.addQueries(ServerListBuilder.java:94)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.ServerListBuilder.add(ServerListBuilder.java:60)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.ServerListBuilder.add(ServerListBuilder.java:53)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	at com.googlecode.jmxtrans.ConfigurationParser.parseServers(ConfigurationParser.java:57)
INFO   | jvm 1    | 2017/11/12 04:43:59 | 	... 11 more
STATUS | wrapper  | 2017/11/12 04:44:01 | <-- Wrapper Stopped
```

--- a/pom.xml
+++ b/pom.xml
@@ -131,7 +131,7 @@
 	</distributionManagement>
 	<properties>
 		<animal-sniffer.version>1.16</animal-sniffer.version>
-		<aws.java.sdk.version>1.11.192</aws.java.sdk.version>
+		<aws.java.sdk.version>1.11.228</aws.java.sdk.version>
 		<!--
 			Maximum number of checkstyle violations allowed. Build will fail
 			if there are more than this number of violations. Feel free to

